src/ext/serial.cpp

```diff
@@ -141,5 +221,82 @@ void ThingSetStream::process_input()
             buf_req[req_pos++] = c;
         }
     }
+
+    #endif /* MBED */
+}
+
+#ifdef __ZEPHYR__
+
+void ThingSetStream::uart_callback(struct device *uart_dev)
+{
+    u8_t c;
+    static int tx_data_idx;
+
+    // Start processing interrupts in ISR.
+    // This function should be called the first thing in the ISR.
+    if (!uart_irq_update(uart_dev)) {
+        // The function output should always be 1.
+        return;
+    }
+
+    /* UART Tx handler:
+    * Note that TX IRQ may be disabled, but uart_irq_tx_ready() may
+    * still return true when ISR is called for another UART interrupt,
+    * hence additional check for tx_data_idx < DATA_SIZE.
+    */
+    // Check if UART TX buffer can accept a new char.
+    if (uart_irq_tx_ready(uart_dev) && tx_data_idx < len_ts) {
+        /* We arrive here by "tx ready" interrupt, so should always
+        * be able to put at least one byte into a FIFO.
+        */
+        if (uart_fifo_fill(uart_dev, (u8_t *)&buf_resp[tx_data_idx++], 1) > 0) {
+            // Character transmitted
+        }
+
+        if (tx_data_idx == len_ts) {
+            /* If we transmitted everything, stop IRQ stream,
+            * otherwise main app might never run.
+            */
+            if(uart_fifo_fill(uart_dev, (u8_t *)&new_line, 1) > 0) {
+                uart_irq_tx_disable(uart_dev);
+                uart_irq_rx_enable(uart_dev);
+                tx_data_idx = 0;
+                len_ts = 0;
+            }
+        }
+    }
+
+    //If data is available to read.
+    if ((uart_irq_rx_ready(uart_dev)) && (command_flag == false)) {
```

Comment:
Is the following code not almost exactly the same of what we have in `process_input` function? So why not use that function and replace `stream->getc` with `uart_fifo_read` in case of Zephyr?